bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GroupItem.java

```diff
@@ -263,9 +263,15 @@ public void stateUpdated(Item item, State state) {
         }
     }
 
+    @Override
     public void setState(State state) {
         State oldState = this.state;
-        this.state = state;
+        if (baseItem != null) {
+            baseItem.setState(state);
+            this.state = baseItem.getState();

```

Comment:
Here we set the state of the base item, which should be responsible for converting the type to the right format, then use that state for the group item.